commisery/checking.py

```diff
@@ -34,25 +35,21 @@ def check_commit(commit, config: Configuration):
             message = file.read()
 
     except IOError:
-        commit = subprocess.check_output(("git", "rev-parse", commit))[:-1].decode(
-            "UTF-8"
-        )
-        message = subprocess.check_output(
-            ("git", "show", "-q", "--format=%B", commit, "--")
-        )[:-1].decode("UTF-8")
-
-    commit_message = CommitMessage.from_message(message)
+        commit = subprocess.check_output(("git", "rev-parse", commit))[:-1].decode("UTF-8")
+        message = subprocess.check_output(("git", "show", "-q", "--format=%B", commit, "--"))[:-1].decode("UTF-8")
+
+    commit_message = parse_commit_message(message)
+
     commit_message.hexsha = commit
 
     return validate_commit_message(commit_message, config)
 
 
-def validate_commit_message(message: CommitMessage, config: Configuration):
+def validate_commit_message(message: typing.Union[CommitMessage, ConventionalCommit], config: Configuration):
```

Comment:
Any reason to not use `COMMIT_TYPE` from rules.py here?

Reply:
Nope.
-> Added COMMIT_TYPE to `commit.py`.